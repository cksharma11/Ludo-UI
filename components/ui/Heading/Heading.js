import React from 'react';
import PropTypes from 'prop-types';
import HeaderStyles from '../../Header/Header.style';

const Heading = ({ content, weight }) => {
  return (
    <div className={weight}>
      {content}
      <style jsx>{HeaderStyles}</style>
    </div>
  );
};

Heading.defaultProps = {
  content: '',
  weight: 'h1'
};

Heading.propTypes = {
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  weight: PropTypes.string
};

export default Heading;
