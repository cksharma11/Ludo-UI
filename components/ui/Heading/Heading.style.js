import css from 'styled-jsx/css';

const HeadingStyles = css`
  .h1 {
    font-size: 20px;
    font-weight: 900;
  }
`;

export default HeadingStyles;
