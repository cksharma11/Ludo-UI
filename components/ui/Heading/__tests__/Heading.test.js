import React from 'react';
import { shallow } from 'enzyme';
import Heading from '../Heading';

describe('Heading', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Heading />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
