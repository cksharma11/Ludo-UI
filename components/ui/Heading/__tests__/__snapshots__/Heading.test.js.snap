// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render properly 1`] = `
"<div className=\\"jsx-4246811236 h1\\">
  <JSXStyle id=\\"4246811236\\">
    .
    h
    e
    a
    d
    e
    r
    .
    j
    s
    x
    -
    4
    2
    4
    6
    8
    1
    1
    2
    3
    6
    {
    f
    o
    n
    t
    -
    f
    a
    m
    i
    l
    y
    :
    f
    a
    n
    t
    a
    s
    y
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    0
    p
    x
    ;
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    w
    i
    d
    t
    h
    :
    1
    0
    0
    %
    ;
    h
    e
    i
    g
    h
    t
    :
    2
    0
    0
    p
    x
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    2
    2
    2
    8
    3
    1
    ;
    c
    o
    l
    o
    r
    :
    #
    d
    6
    5
    a
    3
    1
    ;
    }
  </JSXStyle>
</div>"
`;
